### ISSUE TITLE
`list-style`: no multi line support anymore for if statements

### ISSUE THREAD

**Issue by susnux** [eslint-stylistic/eslint-stylistic#1142]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Since https://github.com/eslint-stylistic/eslint-stylistic/pull/1119 you can no longer have multi line if statements like this:

```js
if (
	node.callee.type !== 'Identifier'
	|| (node.callee.name !== 't' && node.callee.name !== 'n')
) {
	// ...
}
```

>   error  Should not have line break(s) between ')' and ')'  @stylistic/exp-list-style

### Reproduction

can provide a PR with a test if needed

### System Info

```shell
System:
    OS: Linux 6.19
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 24.13.1
    npm: 11.8.0
  npmPackages:
    @stylistic/eslint-plugin: ^5.9.0 => 5.9.0 
    eslint: ^9.39.2 => 9.39.2
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by susnux** [comment]
Before that change I would rather use this:
```js
if (node.callee.type !== 'Identifier'
	|| (node.callee.name !== 't' && node.callee.name !== 'n')
) {
	// ...
}
```

But I can see that this is considered inconsistent for this rule, but the format mentioned in the bug report should still be valid.
I do not want to use the auto-fix that is provided, because it melts the IF-body with the IF-condition visually

```js
if (node.callee.type !== 'Identifier'
	|| (node.callee.name !== 't' && node.callee.name !== 'n')) {
	// code on this line looks visually like it belongs to the if condition
}
```
